src/toil/lib/docker.py

```diff
@@ -192,7 +198,7 @@ def toil_job(job):
     assert deferParam in (None, FORGO, STOP, RM), \
         'Please provide a valid value for deferParam.'
 
-    client = docker.from_env(version='auto')
+    client = docker.from_env(version='auto', timeout=timeout)
```

Comment:
There appear to be two other `client = docker.from_env(version='auto')` calls in this file.  Lines 293 and 328.  I know they don't call the run command so it shouldn't currently affect their behavior by adding this, but it might be good to adjust them as well in case someone adds something in the future and gets confusing behavior since we have to instantiate multiple clients because of pickling.